qadence/backend.py

```diff
@@ -228,9 +236,15 @@ def check_observable(obs_obj: Any) -> AbstractBlock:
                 conv_obs.append(c_obs)
 
             def embedding_fn_dict(a: dict, b: dict) -> dict:
-                embedding_dict = circ_embedding_fn(a, b)
-                for o in obs_embedding_fn_list:
-                    embedding_dict.update(o(a, b))
+                if "circuit" in b:
+                    embedding_dict = {"circuit": circ_embedding_fn(a, b), "observables": dict()}
+                    if "observables" in b:
+                        for o in obs_embedding_fn_list:
+                            embedding_dict["observables"].update(o(a, b))
+                else:
```

Comment:
Do I understand this clause as being "if the passed dict of parameters is not a dict of dicts" ? In that case, I'd make it more explicit because what if a user wants to differentiate over hamiltonian parameters only ?